File: app/models/insight.py
from beanie import Document, PydanticObjectId
from pydantic import BaseModel
from typing import Optional

class Insight(Document):
    # --- Dynamic Data ---
    personName: str
    description: str  
    image: Optional[str] = None
    
    # NEW FIELD: Instagram Profile Link
    insta_link: Optional[str] = None 
    
    # --- Fixed / Styling Data ---
    title: str = "Insight Series"
    bgColor: str = "rgba(125, 212, 238, 0.15)"
    borderColor: str = "#7dd4ee"
    size: str = "regular"
    year:Optional[str]="Alumni"
    class Settings:
        name = "insights"

# Schema for CREATING
class InsightCreate(BaseModel):
    personName: str
    description: str
    image: Optional[str] = None
    insta_link: Optional[str] = None 
    
    # ✅ ADDED: Missing styling fields
    title: Optional[str] = "Insight Series"
    bgColor: Optional[str] = "rgba(125, 212, 238, 0.15)"
    borderColor: Optional[str] = "#7dd4ee"
    size: Optional[str] = "regular"

# Schema for UPDATING
class InsightUpdate(BaseModel):
    personName: Optional[str] = None
    description: Optional[str] = None
    image: Optional[str] = None
    insta_link: Optional[str] = None 
    
    # ✅ ADDED: Missing styling fields
    title: Optional[str] = None
    bgColor: Optional[str] = None
    borderColor: Optional[str] = None
    size: Optional[str] = None

File: src/main/java/com/example/websitebackend/model/Insight.java
package com.example.websitebackend.model;

import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.mongodb.core.mapping.Field;

@Data
@Document(collection = "insights")
public class Insight {

    @Id
    private String id;

    private String personName;
    private String description;
    private String image;

    @Field("insta_link")
    private String instaLink;

    // Fixed / Styling Data defaults
    private String title = "Insight Series";
    private String bgColor = "rgba(125, 212, 238, 0.15)";
    private String borderColor = "#7dd4ee";
    private String size = "regular";
    private String year = "Alumni";
}